src/components/CustomScrollView/useTrackerVisibility.ts

```diff
@@ -0,0 +1,87 @@
+import * as React from "react";
+import { useTimeout } from "../../hooks/useTimeout";
+
+export interface TrackerOptionsProps {
+  /**
+   * Скрывать ли ползунок скроллбара
+   */
+  hideScrollbar?: boolean;
+  /**
+   * Через какое кол-во миллисекунд ползунок скроллбара скрывается
+   */
+  hideScrollbarDelay?: number;
+}
+
+export interface TrackerVisibilityProps {
+  /**
+   * Отвечает за видимость ползунка
+   */
+  trackerVisible: boolean;
+  /**
+   * Позволяет "запланировать" скрытие ползунка. Первый вызов делает ползунок видимым, затем через
+   * delay миллисекунд скрывает его. Если тайм-аут не успевает сработать, то каждый последующий вызов
+   * функции откладывает скрытие ползунка на delay миллисекунд
+   */
+  queueTrackerVisibility: () => void;
+  /**
+   * Позволяет изменить видимость ползунка
+   * @param isVisible - показать/скрыть
+   * @param force - меняет режим форсированного изменения видимости (trackerVisible не будет реагировать на изменения,
+   * пока данный режим включен)
+   */
+  changeTrackerVisiblity: (isVisible: boolean, force?: boolean) => void;
+}
+
+/**
+ * Хук, который позволяет управлять видимостью ползунка скроллбара.
+ * @param hideScrollbar - скрывать ли ползунок скроллбара
+ * @param delay - через какое кол-во миллисекунд ползунок скроллбара скрывается
+ * @returns Объект, содержащий параметры, которые позволяют управлять видимостью ползунка
+ */
+export const useTrackerVisibility = (
+  hideScrollbar = false,
+  delay = 1500
```

Comment:
```suggestion
  delay = 500
```

на маке скролл исчезает где-то через `500ms`

https://user-images.githubusercontent.com/5850354/176163369-34043f16-4602-48a2-ae84-4dd2587ae205.mov